pkg/resources/poddistribution/poddistribution.go

```diff
@@ -0,0 +1,386 @@
+	rs := &k8appsv1.ReplicaSet{}
+	err := client.Get(ctx, k8sTypes.NamespacedName{Name: objDtls[name], Namespace: objDtls[ns]}, rs)
+	if err != nil {
+		logrus.Errorf("Error getting ReplicaSet %v", err)
+		return nil, err
+	}
+	return rs, nil
+}
+
+func getStatefulSet(ctx context.Context, client k8sclient.Client, objDtls map[string]string) (metav1.Object, error) {
+	ss := &k8appsv1.StatefulSet{}
+	err := client.Get(ctx, k8sTypes.NamespacedName{Name: objDtls[name], Namespace: objDtls[ns]}, ss)
+	if err != nil {
+		logrus.Errorf("Error getting StatefulSet %v", err)
+		return nil, err
+	}
+	return ss, nil
+}
+
+func parseKnn(knn string) map[string]string {
+	objDtls := map[string]string{}
+	s := strings.Split(knn, "/")
+	objDtls[kind] = s[0]
+	objDtls[ns] = s[1]
+	objDtls[name] = s[2]
+	return objDtls
+}
+
+func getNamespaces(nsPrefix string, installType string) []string {
+	namespaces := []string{
+		nsPrefix + "3scale",
+		nsPrefix + "rhsso",
+		nsPrefix + "user-sso",
+	}
+
+	if installType == string(integreatlyv1alpha1.InstallationTypeManagedApi) {
+		namespaces = append(namespaces, nsPrefix+"marin3r")
+	}
+	return namespaces
+}
+
+func ReconcilePodDistribution(ctx context.Context, client k8sclient.Client, nsPrefix string, installType string) {
+	for _, ns := range getNamespaces(nsPrefix, installType) {
+		logrus.Infof("Reconciling Pod Balance in ns %s", ns)
+		unbalanced, err := findUnbalanced(ctx, ns, client)
+		if err != nil {
+			logrus.Errorf("Error getting pods to balance on namespace %s. %v", ns, err)
+		} else {
+			for knn, pods := range unbalanced {
+				objDtls := parseKnn(knn)
+				obj, err := getObject(ctx, client, objDtls)
+				if err != nil || obj == nil {
+					logrus.Errorf("Failed to get object from string %s", knn)
+					continue
+				}
+				balance := verifyBalanceCount(ctx, client, obj)
+				if balance {
+					forceRebalance(ctx, client, objDtls, pods)
+				}
+			}
+		}
+	}
+}
+
+func findUnbalanced(ctx context.Context, nameSpace string, client k8sclient.Client) (map[string][]string, error) {
+	nodes := &corev1.NodeList{}
+	unBalanced := map[string][]string{}
+	if err := client.List(ctx,
+		nodes, &k8sclient.ListOptions{}); err != nil {
+		return unBalanced, err
+	}
+	nodesToZone := map[string]string{}
+	for _, n := range nodes.Items {
+		for _, a := range n.Status.Addresses {
+			if a.Type == corev1.NodeInternalIP {
+				nodesToZone[a.Address] = n.Labels[ZoneLabel]
+				break
+			}
+		}
+	}
+	logrus.Infof("nodes to zone %v", nodesToZone)
+	balance := map[string][]string{}
+	objPods := map[string][]string{}
+	l := &corev1.PodList{}
+	listOpts := []k8sclient.ListOption{
+		k8sclient.InNamespace(nameSpace),
+	}
+	if err := client.List(ctx, l, listOpts...); err != nil {
+		logrus.Errorf("Error getting pod lists %v", err)
+		return unBalanced, err
+	}
+
+	// need to check if there is more than 1 pod of a kind
+	podCount := map[string]int{}
+	logrus.Infof("total pods in ns %s: %d", nameSpace, len(l.Items))
```

Comment:
Can remove this I think it is just noise outside development

Reply:
changed to debug